asapdiscovery-docking/asapdiscovery/docking/scripts/make_docked_complexes_schema_v2.py

```diff
@@ -0,0 +1,118 @@
+"""
+Make complex PDB files for docked SDF files.
+"""
+import argparse
+import multiprocessing as mp
+from functools import partial
+from pathlib import Path
+
+from asapdiscovery.data.openeye import load_openeye_sdfs
+from asapdiscovery.data.schema_v2.complex import Complex
+from asapdiscovery.data.schema_v2.ligand import Ligand
+from asapdiscovery.data.schema_v2.target import Target
+from asapdiscovery.data.utils import (
+    MOONSHOT_CDD_ID_REGEX,
+    MPRO_ID_REGEX,
+    construct_regex_function,
+)
+
+
+def make_docked_complex(docked_fn, xtal_dir, out_name, compound_regex, xtal_regex):
+    print(docked_fn, flush=True)
+
+    # Build regex functions if not already built
+    if isinstance(compound_regex, str):
+        compound_regex = construct_regex_function(compound_regex)
+    if isinstance(xtal_regex, str):
+        xtal_regex = construct_regex_function(xtal_regex)
+
+    try:
+        compound_id = compound_regex(docked_fn.parts[-2])
+    except ValueError:
+        print(f"Couldn't find compound id regex match for {str(docked_fn)}", flush=True)
+        return
+    try:
+        xtal_id = xtal_regex(docked_fn.parts[-2])
+    except ValueError:
+        print(f"Couldn't find xtal id regex match for {str(docked_fn)}", flush=True)
+        return
+
+    all_ligs = [
```

Comment:
not necessarily, since the docking results directory (where the ligands are being loaded from) is not necessarily the same as the Fragalysis directory (and in fact should probably necessarily be different)

Reply:
Ah ok that makes sense, perhaps will be simplified by `StructureDir` factory upcoming in #523 but lets cross that bridge when we come to it.